orangecontrib/spectroscopy/widgets/owpeakfit.py

```diff
@@ -255,10 +265,27 @@ def progress_interrupt(_: float):
         model_result = {}
         x = getx(data)
         if data is not None and model is not None:
+
             for row in data:
                 progress_interrupt(0)
-                model_result[row.id] = model.fit(row.x, parameters, x=x)
-
+                res = pool.schedule(pool_fit2, (row.x, model.dumps(), parameters, x))
```

Comment:
More for my own learning: my understanding here is that each preview spectrum is running in it's own task. 

1. Does the for loop block on each `while not res.done()` or is that handled by the `concurrent.futures.wait()`. I suppose it's a moot point as there is only 1 preview worker anyway.
2. Is this really efficient? I think that for 3-10 fits the start-up time might dominate, but maybe this is what we were discussing earlier in the PR and it's not a big deal

Reply:
1 `.done()` should not block, so the only blocking call is `wait()`.
2. There is no start-up time, except for the first fit, OR when previews preview could not be canceled nicely. If the optimization function used by `lmfit` was not so stubbornly written, we could ALWAYS stop it nicely. Oh yes, but then we would not even need separate processes, so... What is here is certainly a compromise. I